### PR TITLE
Reducers and actions

### PR DESCRIPTION
**Pull request by Marcraphael12**
In this PR, we are creating reducers for our app
##Project requirements
- Write your book's actions and reducer (in `/src/redux/books/books.js`) using the [*ducks* pattern](https://github.com/erikras/ducks-modular-redux). 
    - Define *action types* for adding and removing a book.
    - Set the initial state to be an empty array of books.
    - Export *Action Creators* for your *actions*.
    - Write your reducer and export it as default.
        - Define state changes for the actions that you created.
        - In case of unknown action - return the current state.
- Write your categories actions and reducer (in `/src/redux/categories/categories.js`) using the [*ducks* pattern](https://github.com/erikras/ducks-modular-redux). 
    - Define an *action type* for checking the status.
    - Set the initial state to be an empty array of categories.
    - Export *Action Creators* for your *actions*.
    - Write your reducer and export it as default.
        - For the check status action return a string "Under construction".
        - In case of unknown action - return the current state.

- Configure the Redux Store (`/src/redux/configureStore.js`):
    - Importing the necessary methods from Redux Toolkit.
    - Combine both reducers into a root reducer by using the `configureStore` function.

### CODE REVIEW

File: src/components/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// action types
+const ADD_BOOK = 'bookStore/books/ADD_BOOK';
```
Comment (Kamba56) [inline src/components/redux/books/books.js]: - [x] Greta job implementing the books redux. Kindly note that your redux folder should ne outside the components folder.

- [x] kindly add checkstatus action in the categories.js file. The action should be of type: CHECKSTATUS and and should return a string "under obstruction". Also add a categoriesReducer.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @Marcraphael12, 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great use of Github workflow!👏👏👏
- [x] All linters passed

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/redux/books/books.js (hunk 1 above)
Comment (Marcraphael12) [reply src/components/redux/books/books.js]: All done... thanks for your review

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// action type
+const CHECK_STATUS = 'CHECK_STATUS';
```
Comment (DammyShittu) [inline src/redux/categories/categories.js]: - [x] Kindly use the ducks pattern for the action type here.

  ```
  const CHECK_STATUS = 'bookstore/categories/CHECK_STATUS';
  ```

File: src/redux/categories/categories.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+});
+
+// initial state
+const initialState = [];
+
+// reducer
+const categoriesReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case CHECK_STATUS:
+      return ['Under construction', action.payload];
```
Comment (DammyShittu) [inline src/redux/categories/categories.js]: - [x] We only need to return the string "Under construction", not an array. And we also do not have a `action.payload`

  ```
  return 'Under construction';
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Marcraphael12,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categories.js (hunk 3 above)
Comment (Marcraphael12) [reply src/redux/categories/categories.js]: All done... very helpful

**Review by gemmen29 — APPROVED** [review]
Hi @Marcraphael12, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**